test/src/menuui/test_intel_parse.cpp

```diff
@@ -0,0 +1,307 @@
+#include <gtest/gtest.h>
+
+#include "menuui/techmenu.h"
+#include "localization/localize.h"
+
+#include "util/FSTestFixture.h"
+
+class IntelParseTest : public test::FSTestFixture {
+ public:
+	IntelParseTest() : test::FSTestFixture(0) {
+		pushModDir("menuui");
+		pushModDir("intel_parse");
+	}
+
+ protected:
+	virtual void SetUp() override {
+		test::FSTestFixture::SetUp();
+
+		// techroom_intel_init() is not here so it is handled in
+		// the test if it throws.
+	}
+	virtual void TearDown() override {
+		// this is here so that it is run even when a test fails
+		techroom_intel_reset();
+
+		test::FSTestFixture::TearDown();
+	}
+};
+
+// Commonly used expected intel data entries.
+static intel_data expected_foo = {
+	"Foo name", // XSTR id 3000
+	"Foo desc", // XSTR id 3001
+	"Foo anim",
+	IIF_IN_TECH_DATABASE | IIF_DEFAULT_IN_TECH_DATABASE
+};
+static intel_data expected_bar = {
+	"Bar name", // XSTR id 3002
+	"Bar desc", // XSTR id 3003
+	"Bar anim",
+	0
+};
+static intel_data expected_baz = {
+	"Baz name", // XSTR id 3004
+	"Baz desc", // XSTR id 3005
+	"Baz anim",
+	IIF_IN_TECH_DATABASE | IIF_DEFAULT_IN_TECH_DATABASE
+};
+
+// use SCOPED_TRACE("<test name>") before calling this
+static void test_intel_data_equal(intel_data& i1, intel_data& i2)
+{
+	EXPECT_STREQ(i1.name, i2.name);
+	EXPECT_STREQ(i1.desc, i2.desc);
+	EXPECT_STREQ(i1.anim_filename, i2.anim_filename);
+	EXPECT_EQ(i1.flags, i2.flags);
+}
+
+/*
+ * These tests assume that the raw file is preprocessed correctly,
+ * i.e. comments are removed correctly, necessary hacking on retail
+ * data is done, ...
+ * Also, XSTR("...", 123) parsing is assumed to be done correctly.
+ * Only parsing of the actual contents is tested.
+ */
+
+// A missing file just means there are no entries.
+TEST_F(IntelParseTest, missing_file) {
+	techroom_intel_init();
+
+	EXPECT_EQ(Intel_info_size, 0);
+}
+
+// The same with an empty file.
+TEST_F(IntelParseTest, empty_file) {
+	techroom_intel_init();
+
+	EXPECT_EQ(Intel_info_size, 0);
+}
+
+// The same with only white space.
+TEST_F(IntelParseTest, only_white_space) {
+	techroom_intel_init();
+
+	EXPECT_EQ(Intel_info_size, 0);
+}
+
+// A single valid entry.
+TEST_F(IntelParseTest, single) {
+	SCOPED_TRACE("single");
+
+	techroom_intel_init();
+
+	ASSERT_EQ(Intel_info_size, 1);
+
+	test_intel_data_equal(expected_foo, Intel_info[0]);
+}
+
+// The same, but with translation to German.
+TEST_F(IntelParseTest, single_translate) {
+	SCOPED_TRACE("single_translate");
```

Comment:
I don't think you need a scoped trace if the message is the same as the unit test name.